### PR TITLE
[WIP] Add draft of Event Sourcing definition

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

File: event-sourcing-basics/new.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+A lot of different terms are often thrown into the mix when discussing event sourcing that are complementary to the pattern, but not always needed. Let's begin with a basic definition.
+
+## Event Sourcing
+
+Event sourcing is persisting changes to application state as an ordered sequence of events. You can then calculate the current application state from replaying the history of events, instead of directly checking it.
+
+{IMAGE}
+
+The pattern is especially useful for distributed systems with multiple asynchronous as it helps you trace what events occurred within the application as a whole.
```
Comment (riccardone) [inline event-sourcing-basics/new.md]: I suggest the following paragraph instead:
The pattern is especially useful to build event-sourced applications as it helps you trace what events occurred within the application as a whole.

Comment (jageall) [inline event-sourcing-basics/new.md]: if we are going to start with a basic definition, I would expand with basic usages rather than jumping straight into the kind of system that people don't normally have to write

In fact, I would start with something along the lines of it giving you a log of changes over time. It removes the question of how we got to the current state, it facilitates full testing of input and output state (given sequence of events, when something makes a state change, expect exactly this sequence of events)

Comment (jageall) [inline event-sourcing-basics/new.md]: asynchronous is not the word you are looking for here. asynchronous means "non-blocking" multiple concurrent or highly concurrent usage is much more appropriate term, concurrent meaning "happening or done at the same time"

File: event-sourcing-basics/new.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Overview
+
+A lot of different terms are often thrown into the mix when discussing event sourcing that are complementary to the pattern, but not always needed. Let's begin with a basic definition.
```
Comment (jageall) [inline event-sourcing-basics/new.md]: it isn't really a basic definition, it is the definition. well, the actual definition is "Capture all changes to an application state as a sequence of events."

File: event-sourcing-basics/new.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Event Sourcing
+
+Event sourcing is persisting changes to application state as an ordered sequence of events. You can then calculate the current application state from replaying the history of events, instead of directly checking it.
+
+{IMAGE}
+
+The pattern is especially useful for distributed systems with multiple asynchronous as it helps you trace what events occurred within the application as a whole.
+
```
Comment (jageall) [inline event-sourcing-basics/new.md]: not sure why these are all in something called event sourcing basics. this is more like a pattern catalogue of other patterns that are complimentary

File: event-sourcing-basics/new.md (hunk 1 above)
Comment (ChrisChinchilla) [reply event-sourcing-basics/new.md]: @riccardone Is this helpful though? As it presumes you know what the term "event-sourced" means and I thought this was supposed to be very high level. And this is under the heading "Event Sourcing", you can't really define a term using the same term.

File: event-sourcing-basics/new.md (hunk 3 above)
Comment (ChrisChinchilla) [reply event-sourcing-basics/new.md]: @jageall Yes, this was mentioned in the discussion around this PR, these are placeholders for complimentary patterns.

File: event-sourcing-basics/new.md (hunk 2 above)
Comment (ChrisChinchilla) [reply event-sourcing-basics/new.md]: @jageall I don't entirely understand your point here, I'll cut "basic", but I don't understand what you want me to do with the second part of the comment here.

File: event-sourcing-basics/new.md (hunk 3 above)
Comment (jageall) [reply event-sourcing-basics/new.md]: I was referring to the name in the document path :)
This, with it's terms/headings is more like a pattern catalogue or pattern language (which is really a fancy way of me saying "glossary of terms with a bit of extra detail").  Which is fine, in fact I would argue pretty critical for the being able to have more in depth documents later. I am quite happy for us to have a doc that captures the related patterns and has pretty concise definitions for them. we can always branch off for more detailed explanations later.

File: event-sourcing-basics/new.md (hunk 1 above)
Comment (ChrisChinchilla) [reply event-sourcing-basics/new.md]: OK @jageall This came from somewhere, might need to check over other words.

File: event-sourcing-basics/new.md (hunk 3 above)
Comment (jageall) [reply event-sourcing-basics/new.md]: maybe a better path would be something event-sourced-applications/getting-started/glossary-of-terms.md

File: event-sourcing-basics/new.md (hunk 2 above)
Comment (jageall) [reply event-sourcing-basics/new.md]: probably my inept attempt at tagging the correct lines. let me try again...

**Comment by ChrisChinchilla** [comment]
@jageall and @riccardone I'm already wary of what is supposed to be a brief sentence or two (at this moment) getting overly complex, but here's my attempt to add a bit more.

bear in mind that you can also use the new GitHub "suggest" feature to make suggestions inline that might illustrate your points better.

File: event-sourcing-basics/new.md (hunk 3 above)
Comment (ChrisChinchilla) [reply event-sourcing-basics/new.md]: This really is just a scratch doc right now, don't pay too much attention to the path or file name etc :)

**Comment by ChrisChinchilla** [comment]
Moving to Google Docs for now, so Stephanie can also be involved
